ui/pages/token-allowance/token-allowance.js

```diff
@@ -0,0 +1,546 @@
+        </Typography>
+      </Box>
+      <Box>
+        <ContractTokenValues
+          tokenName={tokenSymbol}
+          address={tokenAddress}
+          chainId={fullTxData.chainId}
+          rpcPrefs={rpcPrefs}
+        />
+      </Box>
+      <Box
+        marginTop={1}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.ROW}
+        justifyContent={JUSTIFY_CONTENT.CENTER}
+      >
+        <Button
+          type="link"
+          onClick={() => setShowContractDetails(true)}
+          className="token-allowance-container__verify-link"
+        >
+          <Typography variant={TYPOGRAPHY.H6} color={COLORS.PRIMARY_DEFAULT}>
+            {t('verifyContractDetails')}
+          </Typography>
+        </Button>
+      </Box>
+      <Box margin={[4, 4, 3, 4]}>
+        <ReviewSpendingCap
+          tokenName={tokenSymbol}
+          currentTokenBalance={parseFloat(currentTokenBalance)}
+          tokenValue={10}
+          onEdit={() => setIsFirstPage(true)}
+        />
+      </Box>
+      {!isFirstPage && (
+        <Box className="token-allowance-container__card-wrapper">
+          {renderApproveContentCard({
+            symbol: <i className="fa fa-tag" />,
+            title: t('transactionFee'),
+            showEdit: true,
+            showAdvanceGasFeeOptions: true,
+            onEditClick: showCustomizeGasModal,
+            content: renderTransactionDetailsContent(),
+            noBorder: useNonceField || !showFullTxDetails,
+          })}
+        </Box>
+      )}
+      <Box
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.ROW}
+        justifyContent={JUSTIFY_CONTENT.CENTER}
+      >
+        <Button
+          type="link"
+          onClick={() => setShowFullTxDetails(!showFullTxDetails)}
+          className="token-allowance-container__view-details"
+        >
+          <Typography
+            variant={TYPOGRAPHY.H6}
+            color={COLORS.PRIMARY_DEFAULT}
+            marginRight={1}
+          >
+            {t('viewDetails')}
+          </Typography>
+          {showFullTxDetails ? (
+            <i className="fa fa-sm fa-angle-up" />
+          ) : (
+            <i className="fa fa-sm fa-angle-down" />
+          )}
+        </Button>
+      </Box>
+      {showFullTxDetails ? renderFullDetails() : null}
+      <PageContainerFooter
+        cancelText={t('reject')}
+        submitText={isFirstPage ? t('next') : t('approveButtonText')}
+        onCancel={() => handleReject()}
+        onSubmit={() => (isFirstPage ? setIsFirstPage(false) : handleApprove())}
+      />
+      {showContractDetails && (
+        <ContractDetailsModal
+          tokenName={tokenSymbol}
+          onClose={() => setShowContractDetails(false)}
+          tokenAddress={tokenAddress}
+          toAddress={toAddress}
+          chainId={fullTxData.chainId}
+          rpcPrefs={rpcPrefs}
+          origin={origin}
+          siteImage={siteImage}
+        />
+      )}
+    </Box>
+  );
+}
+
```

Comment:
Any chance we could get a story for this component? I know it's going to take a lot of mock data but if it's not too much overhead I think it could add a lot of value!